eng/common/TestResources/New-TestResources.ps1

```diff
@@ -158,9 +159,28 @@ function NewServicePrincipalWrapper([string]$subscription, [string]$resourceGrou
         Write-Warning "Update-Module Az.Resources -RequiredVersion 5.3.1"
         exit 1
     }
-    $servicePrincipal = Retry {
-        New-AzADServicePrincipal -Role "Owner" -Scope "/subscriptions/$SubscriptionId/resourceGroups/$ResourceGroupName" -DisplayName $displayName
+
+    try {
+        $servicePrincipal = Retry {
+            New-AzADServicePrincipal -Role "Owner" -Scope "/subscriptions/$SubscriptionId/resourceGroups/$ResourceGroupName" -DisplayName $displayName
+        }
+    } catch {
+        # The underlying error "The directory object quota limit for the Principal has been exceeded" gets overwritten by the module trying
+        # to call New-AzADApplication with a null object instead of stopping execution, which makes this case hard to diagnose because it prints the following:
+        #      "Cannot bind argument to parameter 'ObjectId' because it is an empty string."
+        # Provide a more helpful diagnostic prompt to the user if appropriate:
+        $apps = Get-AzADApplication -OwnedApplication
+        if ($apps.Length -ge 135) {
```

Comment:
Where is this magic number coming from?

Reply:
I couldn't find any documentation on per-user quotas, but it seems to be the app registration number that I start getting quota errors for, and I can only assume it's a tenant-level setting. I can try to get some others to corroborate to see if this is consistent across our tenant, or we could just always print this warning along with the current user's app count.

Reply:
I would probably start with the later and aways print the count and give perhaps give folks a warning if the number is high enough that this might be the problem.